helpers/ensure-aur-xlcore.sh

```diff
@@ -0,0 +1,58 @@
+. helpers/prompt.sh
+. helpers/error.sh
+
+CHECK_FOR_FLATPAK() {
+    if command -v flatpak &>/dev/null; then
+        # dev.goats.xivlauncher is the current flatpak name for XLCore on flathub. Change this if that name changes for some reason.
+        if flatpak list --app | grep "dev.goats.xivlauncher"; then
+            error "Detected the flatpak version of XLCore. ACT cannot run in a flatpak environment. Please back up and delete your .xlcore directory, uninstall the flatpak, install the AUR/MPR build of XIVLauncher-git, and re-run this script."
+            echo ""
+            exit 1
+        else
+            success "Flatpak installation not present, checking for AUR/MPR build now..."
+            CONTROL_FORK_DISTRO
+        fi
+    else
+        warn "Flatpak is not installed. Continuing under the assumption that you cannot install a flatpak without it."
+        CONTROL_FORK_DISTRO
+    fi
+}
+
+CONTROL_FORK_DISTRO() {
+    if command -v pacman &>/dev/null; then
+        CHECK_FOR_AUR
+    elif command -v apt &>/dev/null; then
+        CHECK_FOR_MPR
+    else
+        error "Your distribution isn't supported by these setup scripts. This script only supports"
+        echo "Arch-based and Debian-based distros, such as Arch Linux/Manjaro and Debian/Ubuntu/Linux Mint"
+    fi
+}
```

Comment:
This should not fail out if they're on an unsupported platform since the scripts don't directly require either platform as far as I can see?

Instead `warn` that they're running on an untested platform and errors may occur, but allow them to continue anyways.